File: php/hanoi_tower.php
<?php
  $count = 0;

  function hanoi($n, $from, $mid, $to) {
    global $count;
    if ($n == 1) {
      $count++;
    } else {
      hanoi($n - 1, $from, $to, $mid);
      $count++;
      hanoi($n - 1, $mid, $from, $to);
    }
  }

  function executeHanoi($n) {
    global $count;
    echo "Starting...\n";
    hanoi($n,'A','B','C');
    echo "$count iterations\n";
  }

  executeHanoi(32);

?>

File: nim/hanoi_tower.nim
proc hanoi(n: int, a: int, b: int, c: int): int =
  if n == 1:
    return 1
  else:
    return hanoi(n-1, a, c, b) + 1 + hanoi(n-1, b, a, c)

echo hanoi(32, 1, 2, 3)